src/lake/Lake/Config/Package.lean

```diff
@@ -217,6 +219,19 @@ hydrate_opaque_type OpaquePackage Package
 instance : Hashable Package where hash pkg := hash pkg.config.name
 instance : BEq Package where beq p1 p2 := p1.config.name == p2.config.name
 
+instance : ToJson Package where
+  toJson x := .mkObj <| .join [
+    [("dir", toJson x.dir)],
+    [("config", toJson x.config)],
+    Json.opt "remoteUrl" x.remoteUrl?,
+    Json.opt "gitTag" x.gitTag?,
+    [("deps", toJson x.depsConfig)],
+    [("libs", toJson x.leanLibConfigs)],
+    [("exes", toJson x.leanExeConfigs)],
+    [("defaultTargets", toJson x.defaultTargets)],
+    [("scripts", toJson x.scripts)],
+    [("defaultScripts", toJson (x.defaultScripts.map (·.name)))]]
```

Comment:
I don't think there is a way to do the equivalent of `serde(skip)` with the `deriving ToJson` handler, so I inlined the autogenerated definition here. An alternative way to write this would be to have `structure PackageForJson where ... <json-safe fields of Package> ... deriving ToJson` and then `instance ToJson Package where toJson pkg := toJson { pkg with : PackageForJson }`.